source/agora/flash/Channel.d

```diff
@@ -2270,6 +2267,59 @@ LOuter: while (1)
         return update_tx;
     }
 
+    /***************************************************************************
+
+        Prepare the settlement TX and publish
+
+        Params:
+            update = update to be published
+
+        Returns:
+            published settlement TX
+
+    ***************************************************************************/
+
+    protected Transaction publishSettlementTx (in UpdatePair update)
+    {
+        auto settle_tx = update.settle_tx.clone();
+        assert(this.last_externalized_update_utxo != Hash.init);
+
+        settle_tx.inputs[0].utxo = this.last_externalized_update_utxo;
+
+        auto utxos = this.getFeeUTXOs(settle_tx.sizeInBytes());
+        settle_tx.inputs ~= utxos.utxos.map!(hash => Input(hash)).array;
+        settle_tx.inputs.sort();
+
+        if (utxos.total_value.sub(utxos.total_fee) && utxos.total_value > 0.coins)
+            settle_tx.outputs[update.multi_settle_sig.output_idx] = Output(utxos.total_value, this.flash_conf.key_pair.address);
+        else
+            // todo: this is a requirement of SigHash.OmitSingle, fix the need to always have a refund output
+            settle_tx.outputs[update.multi_settle_sig.output_idx] = Output(Amount(1), this.flash_conf.key_pair.address);
```

Comment:
I think allowing an `Output` to have an `Amount` of zero (just until the issue for `SigHash.OmitSingle` is resolved) is better than creating Coin units from nothing.

Reply:
We ateş not creating any coins, that is impossible thru user TXs slnce TX validation will catch that. We are only paying less fees, although by just a single unit.